### PR TITLE
Return to iso

### PR DESCRIPTION
**Pull request by JackKelly-Bellroy**

### CODE REVIEW

**Comment by lrworth** [comment]
What was the problem with slashes being a Lens? We've already had it both ways so it'd be useful to write a warning against changing it back to Lens in the future, along with some good reasoning. It's not easy for me to connect Ed Kmett's comment with a reason for this change.

**Comment by JackKelly-Bellroy** [comment]
It's an improper lens anyway, as it violates the law `view l (set l v s) = v` in the same way that the iso did: If you set `[":"]` then read it back out, you get `["", ""]`. The lens is also kinda weird in that if you set through it, that replaces the entire text. Since the pitch for replacing it with a lens was to provide only lawful optics, there is no benefit to doing so.

Iso matches those expectations much more closely - it really is an alternate view on the whole structure, even if it does slightly funny things.

File: src/Network/AWS/ARN.hs [hunk 1]
```diff
@@ -171,35 +169,40 @@ _ARN = prism' renderARN parseARN
 
 -- | Split a 'Text' into colon-separated parts.
 --
--- This is useful for editing the resource part of an ARN:
+-- This is an improper 'Iso'' (@Text.intercalate ":"@ is a retraction
+-- of @Text.splitOn ":"@ but not a section). The 'Iso'' laws are
```
Comment (lrworth) [inline src/Network/AWS/ARN.hs]: The terms "retraction" and "section" might be a bit jarring? This sentence is redundant with the following counterexamples anyway.

Comment (JackKelly-Bellroy) [reply src/Network/AWS/ARN.hs]: Disagree with the redundancy - it precisely describes what goes wrong. How about: `@Text.intercalate ":" . Text.splitOn ":" = id@, but @Text.splitOn ":" . Text.intercalate ":" /= id@`?

Comment (lrworth) [reply src/Network/AWS/ARN.hs]: I think that's better, thanks

**Review by cdaloisio — APPROVED** [review]

**Pull request merged**